autoload/lsp/utils.vim

```diff
@@ -142,6 +142,14 @@ endfunction
 function! lsp#utils#echo_with_truncation(msg) abort
     let l:msg = a:msg
     let l:winwidth = winwidth(0)
+    
+    if &showcmd
+        let l:winwidth -= 11
+    endif
+
+    if &laststatus != 2 && &ruler
+        let l:winwidth -= 18
+    endif
 
     if l:winwidth < strdisplaywidth(l:msg)
```

Comment:
check `l:winwidth > 0` if you do `let l:winwidth -= 11`.

Reply:
Added a check for > 5
If the width is too small, it just outputs it non-truncated and you get the prompt. There's only so much you can do when the width doesn't accommodate the message - leave it up to vim to handle.